libsolc/libsolc.cpp

```diff
@@ -43,17 +43,17 @@ ReadCallback::Callback wrapReadCallback(CStyleReadFileCallback _readCallback = n
 	ReadCallback::Callback readCallback;
 	if (_readCallback)
 	{
-		readCallback = [=](string const& _path)
+		readCallback = [=](string const& _kind, string const& _data)
 		{
 			char* contents_c = nullptr;
 			char* error_c = nullptr;
-			_readCallback(_path.c_str(), &contents_c, &error_c);
+			_readCallback(_kind.c_str(), _data.c_str(), &contents_c, &error_c);
 			ReadCallback::Result result;
 			result.success = true;
 			if (!contents_c && !error_c)
 			{
 				result.success = false;
-				result.responseOrErrorMessage = "File not found.";
+				result.responseOrErrorMessage = "Callback not found.";
```

Comment:
Hm, not sure this is the right way to do it. Perhaps we should let empty `contents_c` pass through as valid (it could be, depending on the kind) and force the callback to return `File not found` if that's the case.

Reply:
But isn't `File not found` relevant only for the import callback?

Reply:
Exactly, and this is a generic piece of code.

Reply:
Ah OK we can mandate to return an empty string (still an allocated pointer).

Reply:
Wait, what case would that be? If content is valid but empty? Can you write the new rules here?

Reply:
@axic any update required here? I'm still not sure I understood it

Reply:
I'm confused myself.

Reply:
Moved this to #7769.